ttnn/conv.py

```diff
@@ -0,0 +1,137 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import math
+import pathlib
+from typing import Optional, Tuple, Union, Dict
+
+
+import tt_lib as ttl
+
+from ttnn.tensor import (
+    Shape,
+    Tensor,
+    Device,
+    from_torch,
+    to_torch,
+    to_device,
+    from_device,
+    to_layout,
+    DataType,
+    MemoryConfig,
+    MathFidelity,
+    DRAM_MEMORY_CONFIG,
+    Layout,
+    ROW_MAJOR_LAYOUT,
+    TILE_LAYOUT,
+    TILE_SIZE,
+    has_storage_type_of,
+)
+
+from tt_eager.tt_dnn.op_library.sliding_window_op_infra.tt_py_composite_conv import (
+    TTPyCompositeConv,
+    SlidingWindowOpParams,
+)
+
+
+class Conv2D:
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+        kernel_size: Union[int, Tuple[int, int]],
+        stride: Union[int, Tuple[int, int]] = 1,
+        padding: Union[int, Tuple[int, int]] = 0,
+        dilation: Union[int, Tuple[int, int]] = 1,
+        groups: int = 1,
+        padding_mode: str = "zeros",
+        dtype: DataType = None,
+        *,
+        device: Device,
+        use_1d_systolic_array: bool,
+        batch_size: int,
+        input_height: int,
+        input_width: int,
+        reader_patterns_cache: Dict,
+        weight: Tensor,
```

Comment:
There should be no positional args after *, since * will catch everything.

Reply:
Arguments with no default values after * are required args that user have to set.